include/utilities/EndianCopy.hpp

```diff
@@ -0,0 +1,58 @@
+#ifndef NGEN_ENDIANCOPY_H
+#define NGEN_ENDIANCOPY_H
+
+#include <vector>
+
+#include <boost/endian.hpp>
+
+namespace utils {
+
+/**
+ * @brief
+ * Copies bytes from @param{src} to @param{dst},
+ * converts the endianness to native,
+ * and increments @param{index} by the number of bytes
+ * used to store @tparam{S}
+ * 
+ * @tparam S a primitive type
+ * @param src a vector of bytes
+ * @param index an integral type tracking the starting position of @param{dst}'s memory
+ * @param dst output primitive
+ * @param order endianness value (0x01 == Little; 0x00 == Big)
+ */
+template<typename S>
+void copy_from(const std::vector<uint8_t>& src, int& index, S& dst, uint8_t order)
+{
+    std::memcpy(&dst, &src[index], sizeof(S));
```

Comment:
```suggestion
    std::memcpy(&dst, &src[index], sizeof(dst));
```
This spelling doesn't require the reader to look up and see that `decltype(dst)` is `S`

Reply:
Also, I think this should preferentially be written
```suggestion
    std::memcpy(&dst, src.data() + index, sizeof(dst));
```
to emphasize that we're not actually subscripting the vector for its element at `index`